addons/im_livechat/controllers/main.py

```diff
@@ -190,7 +190,7 @@ def get_session(self, channel_id, anonymous_name, previous_operator_id=None, cha
                 'name': channel_vals['name'],
                 'chatbot_current_step_id': channel_vals['chatbot_current_step_id'],
                 'state': 'open',
-                'operator_pid': (operator_partner.id, display_name.replace(',', '')),
+                'operator_pid': {'name': display_name, **operator_partner.mail_partner_format(fields={'id': True, 'write_date': True})[operator_partner]},
```

Comment:
The most appropriate would be to use `_get_partner_data` from the member, it already includes `user_livechat_username` stuff, this would remove code duplication, and prevents from sending "fake" name here, mixing up different data.

Reply:
Ok actually it won't be possible because when not persisted there is no member... This is rather annoying, I guess ok to keep in the context of this PR, but the duplication on `user_livechat_username` logic is not great.